lib/http3/qpack.c

```diff
@@ -1279,6 +1279,8 @@ static h2o_iovec_t finalize_flatten(struct st_h2o_qpack_flatten_context_t *ctx)
 
     /* prepend frame header */
     size_t len_len = quicly_encodev_capacity(ctx->headers_buf.size - start_off);
+    if (serialized_header_len != NULL)
+        *serialized_header_len = ctx->headers_buf.size - start_off;
```

Comment:
```suggestion

    if (serialized_header_len != NULL)
        *serialized_header_len = ctx->headers_buf.size - start_off;

    /* prepend frame header */
    size_t len_len = quicly_encodev_capacity(ctx->headers_buf.size - start_off);
```

I think the calculation logic is correct. As it does not depend on the size of the frame header, it might make sense to calculate the size of the response headers before working on the frame header.